mlrun/k8s_utils.py

```diff
@@ -108,19 +108,33 @@ def clean_pods(self, namespace=None, selector="", states=None):
         for item in items:
             self.del_pod(item.metadata.name, item.metadata.namespace)
 
-    def create_pod(self, pod):
+    def create_pod(self, pod, max_retry=3, retry_interval=5):
         if "pod" in dir(pod):
             pod = pod.pod
         pod.metadata.namespace = self.resolve_namespace(pod.metadata.namespace)
-        try:
-            resp = self.v1api.create_namespaced_pod(pod.metadata.namespace, pod)
-        except ApiException as exc:
-            logger.error(f"spec:\n{pod.spec}")
-            logger.error(f"failed to create pod: {exc}")
-            raise exc
 
-        logger.info(f"Pod {resp.metadata.name} created")
-        return resp.metadata.name, resp.metadata.namespace
+        retry_count = 0
+        while True:
+            try:
+                resp = self.v1api.create_namespaced_pod(pod.metadata.namespace, pod)
+            except ApiException as exc:
+                logger.error(f"spec:\n{pod.spec}")
+                logger.error(f"failed to create pod: {exc}")
+
+                # known k8s issue, see https://github.com/kubernetes/kubernetes/issues/67761
+                if "gke-resource-quotas" in str(exc) and retry_count < max_retry:
+                    logger.warning(
+                        "failed to create pod due to gke resource error, "
+                        f"sleeping {retry_interval} seconds and retrying"
+                    )
+                    retry_count += 1
+                    time.sleep(retry_interval)
+                    continue
```

Comment:
I am quite afraid from sleeping on an api requests this can drawback our performance..., what about retrying using client? or maybe retrying without a sleep?